src/output_influx.c

```diff
@@ -327,13 +337,25 @@ static void R_API_CALLCONV print_influx_data(data_output_t *output, data_t *data
 
     // write tags
     while (data) {
-        if (!strcmp(data->key, "model")
-                || !strcmp(data->key, "time")) {
+        if ((!strcmp(data->key, "model") && strstr(influx->metric_format, "[model"))
+            || (!strcmp(data->key, "type") && strstr(influx->metric_format, "[type"))
+            || (!strcmp(data->key, "subtype") && strstr(influx->metric_format, "[subtype"))
+            || (!strcmp(data->key, "hostname") && strstr(influx->metric_format, "[hostname"))
+            || (!strcmp(data->key, "channel") && strstr(influx->metric_format, "[channel"))
+            || (!strcmp(data->key, "id") && strstr(influx->metric_format, "[id"))
+            || (!strcmp(data->key, "protocol") && strstr(influx->metric_format, "[protocol"))) {
+            // this field is already encoded in the metric name -> skip
+        } else if (!strcmp(data->key, "model") && !influx->metric_format[0]) {
+            // non-configurable metric format uses "model" for metric name -> skip
+        } else if (!strcmp(data->key, "time")) {
             // skip
         }
         else if (!strcmp(data->key, "type")
                 || !strcmp(data->key, "subtype")
+                || !strcmp(data->key, "protocol")
                 || !strcmp(data->key, "id")
+                || !strcmp(data->key, "dst_id")
```

Comment:
`dst_id` is not metadata it's part of the Somfy remote commands, there is no `src_id`?

Reply:
You're right that these are not metadata, but to me, the real question is whether this is something that I, as a consumer of these data, should be filtering on, or whether it's some runtime data that I want to plot. In InfluxDB terms (cf. [line protocol](https://docs.influxdata.com/influxdb/v2.0/reference/syntax/line-protocol/)), these "filterable" items correspond to tags, in [Prometheus](https://prometheus.io/docs/practices/naming/) these are called labels.

I don't use Somfy-IOC myself (it's just some neighbor's roller shutters that I hear clearly in this apartment building), but the data looked pretty static and I was thinking that it makes more sense to produce separate time series based on the (source, destination) pairs. That way users can track communication of the individual device pairs as separate series (if they choose to).

I'll move this to a separate patch, though, and if you think that it's still a wrong idea, I'm OK with dropping it as well.

Reply:
It just struck me as pretty arbitrary to filter this one random key. It might look similar to the "id" key but is really just a random data field like all the others. So I assumed an understandable mixup. Yes, let's discuss (universal) handling in another PR.